subt/main.py

```diff
@@ -203,9 +214,14 @@ def follow_wall(self, radius, right_wall=False, timeout=timedelta(hours=3), dist
                     desired_direction = follow_wall_angle(self.scan, radius=radius, right_wall=right_wall)
                     self.go_safely(desired_direction)
                 if dist_limit is not None:
-                    if dist_limit < self.traveled_dist - start_dist:
+                    if dist_limit < abs(self.traveled_dist - start_dist):  # robot can return backward -> abs()
```

Comment:
If the robot drives backward, the value of `self.traveled_dist` would approach `0` so I guess this wouldn't work.

Reply:
the traveled distance is set to zero at the beginning of "follow_wall" (as far as I remember), but the robot moves backward so it goes from 0 to some negative value. This was result of Robik test - I could look it up if you want?
(well, maybe it should not be `self.` variable)
Edit:
`        start_dist = self.traveled_dist`
and the `self.traveled_dist` is signed (based on the fact if you move forward or backward)